b64ct/src/lib.rs

```diff
@@ -200,7 +221,18 @@ pub const fn decoded_len(bytes: &str) -> usize {
 }
 
 const fn decoded_len_inner(n: usize) -> usize {
-    (n * 3) / 4
+    #[cfg(target_pointer_width = "64")]
+    type DoubleWord = u128;
+    #[cfg(target_pointer_width = "32")]
+    type DoubleWord = u64;
+    #[cfg(target_pointer_width = "16")]
+    type DoubleWord = u32;
+
+    if n <= usize::MAX / 3 {
+        (3 * n) / 4
+    } else {
+        ((3 * (n as DoubleWord)) / 4) as usize
```

Comment:
I think it should be possible to do it in a branch-less fashion without introducing the `DoubleWord` type...